heron/downloaders/src/java/org/apache/heron/downloader/DownloadRunner.java

```diff
@@ -38,7 +184,14 @@ public static void main(String[] args) throws Exception {
       file.mkdirs();
     }
 
-    final Downloader downloader = Registry.get().getDownloader(topologyLocation);
+    Map<String, Class<? extends Downloader>> downloaders = new HashMap<>();
+    for (Map.Entry<String, Object> e
+        : ((Map<String, Object>) config.get(Key.DOWNLOADER_PROTOCOLS)).entrySet()) {
+      Class clazz = Class.forName((String) e.getValue());
+      downloaders.put(e.getKey(), clazz);
+    }
+
+    final Downloader downloader = Registry.getDownloader(downloaders, topologyLocation);
```

Comment:
how about only creating the downloader that will be used in the topology? Others seem not necessary to be created.

Reply:
updated